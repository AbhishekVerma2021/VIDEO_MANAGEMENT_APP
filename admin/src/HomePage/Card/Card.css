.Frame {
    height: 40vh;
    width: 100%;
}



#frame {
    height: 100%;
    width: 100%;
    border-radius: 1vh 1vh 0 0;
}

.videoDetails {
    margin: 1% auto 1% auto;
    width: 93%;
}

.videoTitle {
    margin-top: 2%;
    font-size: 1.5vw;
    font-weight: 600;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
}

.videoDescription {
    margin-top: 3%;
    font-size: 1vw;
    width: 100%;
    height: 6vh;
    word-wrap: break-word;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
}

.videoDate {
    margin-top: 3%;
    font-size: 0.9vw;
}


.mainCard {
    position: relative;
    z-index: 0;
    width: 30vw;
    color: black;
    margin: 0 0 0 0.8vw;
    display: flex;
    background-color: rgba(255, 255, 255, 0.429);
    justify-content: space-between;
    flex-direction: column;
    align-items: flex-start;
    gap: 0;
    border-radius: 1vh;
    box-shadow: 5px 5px 5px rgb(200, 200, 200);
    cursor: pointer;
    transition: all 0.3s;
    height: 60vh;
}

.mainCard:hover {
    transform: scale(1.01);
}

.editDeleteBtns {
    display: flex;
    justify-content: space-between;
    gap: 10%;
    align-items: center;
    width: 28vw;
    margin: 3% auto 0 auto;
}

.editDeleteIcons {
    display: flex;
    justify-content: flex-end;
    flex-direction: row;
    gap: 3vw;

}

.actionBtns {
    border: none;
    cursor: pointer;
    background-color: antiquewhite;
    padding: 1% 2% 1% 2%;
    transition: all 0.3s;
    border-radius: 1vw;
}


#viewButton {
    text-align: center;
    border: none;
    cursor: pointer;
    background-color: rgba(8, 0, 255, 0.684);
    padding: 1vh 1vw 1vh 1vw;
    transition: all 0.3s;
    border-radius: 0.5vh;
    text-decoration: none;
    font-weight: 600;
    width: 8.5vw;
    height: 5vh;
    color: rgb(255, 255, 255);
}


#viewButton:hover {
    background-color: rgba(8, 0, 255, 0.497);
    color: white;
}

#delButton {
    margin-top: 1vh;
    font-size: 2.3vw;
    background-color: rgba(105, 105, 105, 0);
    color: rgb(196, 0, 0);
    padding: 0 1vw 0 1vw;
}

#delButton:hover {
    color: red;
}

#editButton {
    font-size: 2.3vw;
    background-color: rgba(105, 105, 105, 0);
    padding: 0 1vw 0 1vw;
    color: darkblue;

}

#editButton:hover {
    color: rgb(22, 64, 203);
}

@media only screen and (max-width: 400px) {

    .mainCard {
        width: 100%;
    }

    .videoDate {
        margin-top: 1vw;
        font-size: 2.5vw
    }

    .videoDesc {
        font-size: 2.5vw;
    }

    .videoTitle {
        font-size: 4vw;
    }

    .editDeleteBtns {
        width: 90%;
    }

    .vidDetails {
        padding: 2vw;
    }


    #editButton {
        padding: 2vw 4vw 2vw 4vw;
        font-size: 4vw;
    }

    #delButton {
        padding: 2vw 4vw 2vw 4vw;
        font-size: 4vw;
    }
}