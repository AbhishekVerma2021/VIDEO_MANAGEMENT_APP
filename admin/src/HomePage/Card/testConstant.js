export const data = {
    "link": "https://www.youtube.com/watch?v=e64JTo7LMmY",
    "title": "This is a title",
    "description": "sadn asd sad  saf rewef wefew f wef we fw ef wef we f",
    "_id": "643d8663cdcd1ac35198e3ae",
    "date": "2023-04-17"
}