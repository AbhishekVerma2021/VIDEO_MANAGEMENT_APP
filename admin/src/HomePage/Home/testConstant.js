export const postCardData = [{
    "date": "2023-04-18T15:29:54.942Z",
    "description": "Unleash your inner power and unlock your full potential to achieve your dreams and goals.",
    "link": "https://www.youtube.com/watch?v=1jhY2fhAsh8",
    "title": "Believe in Yourself, Achieve Your Dreams",
    "_id": "643e66dc66f6cc577576d0aa"
}]

export const searchResultPostDetails = [{
    "date": "2023-04-18T15:29:54.942Z",
    "description": "Empower your mindset with positive thinking, mindset shifts, and personal development.",
    "link": "https://www.youtube.com/watch?v=X2X_qbyVU84",
    "title": "Empowering Your Mindset",
    "_id": "643e66dc66f6cc577576d0a9"
}] 