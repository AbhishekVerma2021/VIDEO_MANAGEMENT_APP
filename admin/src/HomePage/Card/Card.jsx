import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './Card.css'
import { MyContext } from '../../NavBar/MyContext'
import NewVideo from '../../NewVideo/NewVideo'
import 'react-toastify/dist/ReactToastify.css';
import DeleteBox from '../DeleteBox/DeleteBox'

class Card extends Component {

  constructor(props) {
    super(props)
    this.state = {
      open: true,
      updateFlag: true,
      formFlag: false,
      setFlag: this.setFlag,
    }
  }
  setFlag = (value) => { this.setState({ formFlag: value }) }

  handleEditclick = () => {
    this.setState({ formFlag: true })
  }

  render() {
    const { link, title, _id, description } = this.props.data
    const { formFlag, deletBoxFlag } = this.state
    let date = new Date(this.props.data.date)
    date = `${date.getDate()}/${date.getMonth() + 1}/${date.getFullYear()}`
    // date = JSON.stringify(date)
    return <>
      <div className="mainCard">
        <div className="Frame">
          <iframe id="frame" src={link && link.toString().replace("watch?v=", "embed/")} title={title} className="videoLink" frameBorder={0} allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" allowFullScreen ></iframe>
        </div>
        <div className="videoDetails">
          <div title={title} className="videoTitle">
            {title}
          </div>
          <div title={description} className="videoDescription">
            {description}
          </div>
          <div className="videoDate">
            Created On: {date}
          </div>
          <div className="editDeleteBtns">

            <Link to={`/viewPost/${this.props.data._id}`} className='viewLink' >
              <button id='viewButton'>
                View Post
              </button>
            </Link>
            <MyContext.Provider value={{ ...this.state, updateFlag: true, setFlag: this.setFlag }}>
              <div id="editButton" className='actionBtns' onClick={() => this.handleEditclick()}>
                <i className="fa-solid fa-pen-to-square"></i>
              </div>
              {formFlag && (<NewVideo _id={_id} link={link} date={date} description={description} title={title} open={this.state.open} />)}
            </MyContext.Provider>
            <button id="delButton" className='actionBtns' onClick={() => {
              this.setState({
                deletBoxFlag: true
              })
            }} >
              <i className="fa-sharp fa-solid fa-trash" ></i>
            </button>
          </div>
        </div>
        {/* {alert(deletBoxFlag)} */}
        {deletBoxFlag && (<DeleteBox
          openBox={true}
          answer={(res) => {
            // alert(res)
            if (res)
              this.props.onDelete(_id)
            this.setState({ deletBoxFlag: false })
          }} />)}
      </div>
    </>
  }
}

export default Card;