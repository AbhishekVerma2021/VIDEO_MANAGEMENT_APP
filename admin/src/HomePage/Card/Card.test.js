import React from 'react';
import Card from './Card';
import { shallow, configure } from 'enzyme';
import Adapter from "@cfaester/enzyme-adapter-react-18";
configure({ adapter: new Adapter() });
import { data } from './testConstant';


describe('Test <Card/>', () => {
    it('Should render Card component', () => {
        const component = shallow(<Card data={data} />);
        const tree = component.debug();
        expect(tree).toMatchSnapshot();
    })

    it('Link of the video should be "https://www.youtube.com/embed/e64JTo7LMmY"', () => {
        const component = shallow(<Card data={data} />);
        const videoLink = data.link.replace("watch?v=", "embed/")
        expect(component.find('iframe').prop('src')).toEqual(videoLink)
    })

    it('Title of the video should be "This is a title"', () => {
        const component = shallow(<Card data={data} />);
        expect(component.find('.videoTitle').text()).toBe(data.title)
    })

    it('Video description of the video should be "sadn asd sad  saf rewef wefew f wef we fw ef wef we f"', () => {
        const component = shallow(<Card data={data} />);
        expect(component.find('.videoDescription').text()).toBe(data.description)
    })

    it('Date of the video should be "Created On: 17/4/2023"', () => {
        const component = shallow(<Card data={data} />);
        expect(component.find('.videoDate').text()).toEqual('Created On: 17/4/2023')
    })

    it('ViewPost route of the video should be "/viewPost/643d8663cdcd1ac35198e3ae"', () => {
        const component = shallow(<Card data={data} />);
        expect(component.find('.viewLink').prop('to')).toBe('/viewPost/643d8663cdcd1ac35198e3ae')
    })
    it('When clicking on edit button it should "Open edit post section"', () => {
        const component = shallow(<Card data={data} />)
        component.find("#editButton").props().onClick();
        expect(component.state("formFlag")).toBe(true);
    })
})




























































// npm test -- Card -u