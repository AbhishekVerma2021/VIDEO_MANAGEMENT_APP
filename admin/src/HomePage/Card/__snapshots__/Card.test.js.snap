// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Card/> Should render Card component 1`] = `
"<Fragment>
  <div className=\\"mainCard\\">
    <div className=\\"Frame\\">
      <iframe id=\\"frame\\" src=\\"https://www.youtube.com/embed/e64JTo7LMmY\\" title=\\"This is a title\\" className=\\"videoLink\\" frameBorder={0} allow=\\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share\\" allowFullScreen={true} />
    </div>
    <div className=\\"videoDetails\\">
      <div title=\\"This is a title\\" className=\\"videoTitle\\">
        This is a title
      </div>
      <div title=\\"sadn asd sad  saf rewef wefew f wef we fw ef wef we f\\" className=\\"videoDescription\\">
        sadn asd sad  saf rewef wefew f wef we fw ef wef we f
      </div>
      <div className=\\"videoDate\\">
        Created On: 
        17/4/2023
      </div>
      <div className=\\"editDeleteBtns\\">
        <Link to=\\"/viewPost/643d8663cdcd1ac35198e3ae\\" className=\\"viewLink\\">
          <button id=\\"viewButton\\">
            View Post
          </button>
        </Link>
        <ContextProvider value={{...}}>
          <div id=\\"editButton\\" className=\\"actionBtns\\" onClick={[Function: onClick]}>
            <i className=\\"fa-solid fa-pen-to-square\\" />
          </div>
        </ContextProvider>
        <button id=\\"delButton\\" className=\\"actionBtns\\" onClick={[Function: onClick]}>
          <i className=\\"fa-sharp fa-solid fa-trash\\" />
        </button>
      </div>
    </div>
  </div>
</Fragment>"
`;
